libyul/YulStack.cpp

```diff
@@ -164,14 +164,32 @@ void YulStack::optimize(Object& _object, bool _isCreation)
 	if (EVMDialect const* evmDialect = dynamic_cast<EVMDialect const*>(&dialect))
 		meter = std::make_unique<GasMeter>(*evmDialect, _isCreation, m_optimiserSettings.expectedExecutionsPerDeployment);
 
+	auto [optimizeStackAllocation, yulOptimiserSteps, yulOptimiserCleanupSteps] = [&]() -> std::tuple<bool, std::string, std::string>
+	{
+		if (!m_optimiserSettings.runYulOptimiser)
+		{
+			// Yul optimizer disabled, but empty sequence (:) explicitly provided
+			if (m_optimiserSettings.yulOptimiserSteps.empty() && m_optimiserSettings.yulOptimiserCleanupSteps.empty())
+				return std::make_tuple(true, "", "");
+			// Yul optimizer disabled, and no sequence explicitly provided (assumes default sequence)
```

Comment:
It's not a function, it's a lambda :), also, the function is called exactly once, so I don't think it warrants having a proper member function. I also think it's more readable in this context, since it's used to deduce optimizer steps, etc. right before they're used.

Reply:
Lambda is still a function :) My point is that this is an unusual pattern that's both more complex and harder to read than the plain way of doing this.

When I see a lambda my first instinct is to skip it, because I want to get to the line that will be executed next. Then I realize that the lambda was called in place (because I missed the tiny `()` at the end) and have to go back. It adds unnecessary friction to reading the code.